src/cryptonote_core/tx_pool.cpp

```diff
@@ -1472,7 +1500,8 @@ namespace cryptonote
     {
       const bool kept = pass == 1;
       bool r = m_blockchain.for_all_txpool_txes([this, &remove, kept](const crypto::hash &txid, const txpool_tx_meta_t &meta, const cryptonote::blobdata *bd) {
-        if (!!kept != !!meta.kept_by_block)
+        const bool kept_by_block = meta.get_relay_method() == relay_method::block;
+        if (!!kept != !!kept_by_block)
```

Comment:
```suggestion
        if (kept != kept_by_block)
```

Reply:
I decided to keep this as-is for whatever reason, but seems reasonable to change.